File: packages/lynx/create-lynx-library/template-native-module/android/src/main/java/__ANDROID_PACKAGE_PATH__/__MODULE_NAME__.java
package __ANDROID_PACKAGE__;

import androidx.annotation.Nullable;
import com.lynx.jsbridge.LynxMethod;
import com.lynx.jsbridge.LynxNativeModule;
import com.lynx.tasm.behavior.LynxContext;
import __ANDROID_PACKAGE__.generated.__MODULE_NAME__Spec;

@LynxNativeModule(name = "__MODULE_NAME__")
public class __MODULE_NAME__ extends __MODULE_NAME__Spec {
  public __MODULE_NAME__(LynxContext context) {
    super(context);
  }

  @Override
  @LynxMethod
  public void setValue(String key, String value) {
  }

  @Override
  @LynxMethod
  @Nullable
  public String getValue(String key) {
    return null;
  }

  @Override
  @LynxMethod
  public void clear() {
  }
}

File: packages/lynx/create-lynx-library/template-native-module/ios/src/__MODULE_NAME__.m
#import "__MODULE_NAME__.h"

@implementation __MODULE_NAME__

+ (NSString *)name {
  return @"__MODULE_NAME__";
}

+ (NSDictionary<NSString *, NSString *> *)methodLookup {
  return @{
    @"setValue" : NSStringFromSelector(@selector(setValue:value:)),
    @"getValue" : NSStringFromSelector(@selector(getValue:)),
    @"clear" : NSStringFromSelector(@selector(clear)),
  };
}

- (void)setValue:(NSString *)key value:(NSString *)value {
}

- (nullable NSString *)getValue:(NSString *)key {
  return nil;
}

- (void)clear {
}

@end
